src/SDK/SmartDetectorsSDK/DetectorDataNotReadyException.cs

```diff
@@ -0,0 +1,132 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="DetectorDataNotReadyException.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// An exception used by the Smart Detector to indicate that the data it need for analysis
+    /// is not ready yet. The Runtime Environment may use <see cref="ShouldRetryAfter"/> to
+    /// reschedule the analysis, but in any case throwing this exception will not register as
+    /// a failure to run the Smart Detector.
+    /// </summary>
+    [Serializable]
+    public class DetectorDataNotReadyException : Exception
```

Comment:
>DetectorDataNotReadyException [](start = 17, length = 29)

Why did you decide not to call it DetectorSnoozeException? Is it too informal?

Reply:
yes

---
In reply to: [243895026](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/38#discussion_r243895026) [](ancestors = 243895026)